### PR TITLE
Add declarative widgets section

### PR DESCRIPTION
**Pull request by parente**
Ref: https://jupyter.hackpad.com/Spring-2016-Dev-Meeting-h0y1TIAWxz1#:h=Declarative-Widgets-/-Widgets

Hold for one more conversation next week to double-check the feasibility and details.

@lbustelo, @jdfreder, @SylvainCorlay

### CODE REVIEW

**Comment by jdfreder** [comment]
This is looking good @parente .  I also added a note in my PR https://github.com/jupyter/roadmap/pull/16 to the ipywidgets section about integrating decl. widgets.

**Comment by damianavila** [comment]
LGTM.

**Comment by jdfreder** [comment]
I pushed a commit which adds details from a conversation between @sylvaincorlay, @lbustelo, @parente , and I

**Comment by parente** [comment]
@jdfreder @SylvainCorlay @lbustelo I made all three of you collaborators on the fork so you can update this PR. Or you can just close this one and push a new one.

File: README.md [hunk 1]
```diff
@@ -145,15 +145,26 @@ each milestone.
...
+## Declarative widgets
+
+### Next
+
+* Decide where to draw the line between Web Components specific functionality and generic code that should live closer to ipywidgets / jupyter-js-widgets to prevent future breakage.
+* Migrate generic, JS modules to jupyter-js-widgets (may be a no-op).
+* Migrate accompanying Python modules to ipywidgets.
+* Graduate the declarativewidgets from the incubator.  Specifically the move jupyter-incubator/declarativewidgets code to jupyter/jupyter-web-components or another appropriately named repo.
+* Investigate with the jupyter/notebook team about upstreaming a generic bower/npm restful API.
+* Move the R code into the IRkernel and the Scala code into Gino's Scala kernel.
```
Comment (lbustelo) [inline README.md]: Although I love the name you gave it, lets change this from `Gino's Scala kernel` to `Apache Toree`.

**Comment by lbustelo** [comment]
I think this looks good and ready to merge

**Comment by damianavila** [comment]
Not ready, there is some merge conflicts to solve...

**Comment by lbustelo** [comment]
Done!

File: README.md [hunk 2]
```diff
@@ -151,14 +151,24 @@ each milestone.
...
 * The backbone widgets provided with jupyter-js-widgets will be moved into their
   own repository or will be removed altogether.
 
+## Declarative widgets
+
+### Next
+
+* Decide where to draw the line between Web Components specific functionality and generic code that should live closer to ipywidgets / jupyter-js-widgets to prevent future breakage.
+* Migrate generic, JS modules to jupyter-js-widgets (may be a no-op).
+* Graduate the declarativewidgets from the incubator.  Specifically the move jupyter-incubator/declarativewidgets code to jupyter/jupyter-web-components or another appropriately named repo.
```
Comment (rgbkrk) [inline README.md:168]: 👍

File: README.md [hunk 3]
```diff
@@ -151,14 +151,24 @@ each milestone.
...
+* Investigate with the jupyter/notebook team about upstreaming a generic bower/npm restful API.
```
Comment (rgbkrk) [inline README.md:169]: 👍

File: README.md [hunk 4]
```diff
@@ -151,14 +151,24 @@ each milestone.
...
+* Determine the best location for kernel specific code. A possible options it to move Python bits under ipywidgets, move the R code into the IRkernel and the Scala code into Apache Toree or related project.
```
Comment (rgbkrk) [inline README.md:170]: 👍 , also part of the static resources section of the roadmap here

**Comment by rgbkrk** [comment]
I'm in favor.

**Comment by minrk** [comment]
Thanks, all!

**Pull request merged**